### PR TITLE
fix(core): fixing the accessibility for the dialog component

### PR DESCRIPTION
**Pull request by Lokanathan-k**
#### Please provide a link to the associated issue.
Closes: https://github.com/SAP/fundamental-ngx/issues/6100

#### Please provide a brief summary of this pull request.

fixing the accessability test cases

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application

Documentation checklist:
- [x] update `README.md`
- [NA] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: libs/core/src/lib/dialog/base/dialog-config-base.class.ts [hunk 1]
```diff
@@ -29,6 +29,9 @@ export class DialogConfigBase<T> implements DynamicComponentConfig {
     /** Position of the dialog. */
     position?: DialogPosition;
 
+    /** Aria Model for teh dialog component element */
```
Comment (platon-rov) [inline libs/core/src/lib/dialog/base/dialog-config-base.class.ts:32]: Typo in word `the`

File: libs/core/src/lib/dialog/base/dialog-content-base.class.ts [hunk 2]
```diff
@@ -1,4 +1,6 @@
 export class DialogContentBase {
+    /** Dialog Title id */
+    id?: string;
```
Comment (platon-rov) [inline libs/core/src/lib/dialog/base/dialog-content-base.class.ts:3]: If this is title id shouldn't be named `titleId`?

File: libs/core/src/lib/dialog/base/dialog-content-base.class.ts [hunk 3]
```diff
@@ -9,6 +11,18 @@ export class DialogContentBase {
     /** Approve Button Label */
     approveButton?: string;
 
+    /** Aria Model for teh dialog component element */
```
Comment (platon-rov) [inline libs/core/src/lib/dialog/base/dialog-content-base.class.ts:14]: Typo in word `the`.

File: libs/core/src/lib/dialog/dialog.component.ts [hunk 4]
```diff
@@ -157,7 +157,6 @@ export class DialogComponent extends DialogBase implements OnInit, OnChanges, Af
     /** @hidden */
     ngAfterViewInit(): void {
         super.ngAfterViewInit();
-        this.dialogWindow.nativeElement.focus();
```
Comment (platon-rov) [inline libs/core/src/lib/dialog/dialog.component.ts:160]: Why it has been removed?

**Review by platon-rov — CHANGES_REQUESTED** [review]

**Review by kavya-b — COMMENTED** [review]
can you verify the netlify url against JAWS once to see if the reported issues are fixed? I randomly checked for Object Based Dialog example and it still reads 
1. whole content 
2. title twice read
on opening dialog

image

File: libs/core/src/lib/dialog/dialog.component.ts (hunk 4 above)
Comment (Lokanathan-k) [reply libs/core/src/lib/dialog/dialog.component.ts:160]: this does not do anything, and not required

File: apps/docs/src/app/core/component-docs/dialog/examples/component-based/dialog-example.component.ts [hunk 5]
```diff
@@ -9,7 +9,7 @@ import { DialogRef } from '@fundamental-ngx/core/dialog';
             </fd-dialog-header>
 
             <fd-dialog-body>
-                <p id="fd-dialog-body-1" role="dialog" style="text-align: justify; margin: 0">
+                <p id="fd-dialog-body-1" role="document" style="text-align: justify; margin: 0">
```
[the remark this answers is not in the document]
Comment (Lokanathan-k) [reply apps/docs/src/app/core/component-docs/dialog/examples/component-based/dialog-example.component.ts]: cannot add if we add it from the component it just reads as "contentInfoRegion" need to specifically specify role as "document"  if its a static content.

File: apps/docs/src/app/core/component-docs/dialog/examples/auto-label/auto-label-dialog-example.component.html [hunk 6]
```diff
@@ -14,13 +14,15 @@ <h1 id="fd-dialog-header-10" fd-title fd-dialog-title>The History of Pineapple</
...
                 (click)="dialog.close('Continue')"
+				ariaLabel="Interested emphasized"
             ></fd-button-bar>
 
             <fd-button-bar
                 fdInitialFocus
                 label="Cancel"
                 [compact]="true"
                 (click)="dialog.dismiss('Cancel')"
+				ariaLabel="Cancel transparent"
```
Comment (InnaAtanasova) [inline apps/docs/src/app/core/component-docs/dialog/examples/auto-label/auto-label-dialog-example.component.html:25]: is the user really interested in the fact that the button is transparent or emphasized?

File: apps/docs/src/app/core/component-docs/dialog/examples/dialog-complex/dialog-complex-example.component.html [hunk 7]
```diff
@@ -31,7 +32,7 @@ <h1 id="fd-dialog-header-3" fd-title>Fresh Market</h1>
         </fd-dialog-header>
 
         <fd-dialog-body>
-            <ul *ngIf="!(dialogRef.onLoading | async)" fd-list [compact]="true">
+            <ul *ngIf="!(dialogRef.onLoading | async)" fd-list [compact]="true" role="document">
```
Comment (InnaAtanasova) [inline apps/docs/src/app/core/component-docs/dialog/examples/dialog-complex/dialog-complex-example.component.html:35]: `ul` element with role="document"? Why is this needed?

File: apps/docs/src/app/core/component-docs/dialog/examples/auto-label/auto-label-dialog-example.component.html (hunk 6 above)
Comment (Lokanathan-k) [reply apps/docs/src/app/core/component-docs/dialog/examples/auto-label/auto-label-dialog-example.component.html:25]: yes, it has been raised as issues by the accessibility team. the changes has been introduced post verifying with the team.

Comment (kavya-b) [reply apps/docs/src/app/core/component-docs/dialog/examples/auto-label/auto-label-dialog-example.component.html:25]: in that case we should handle this from library itself, right? where `aria-label` can be set to the `ariaLabel` value and the button's `type`, so that user is not expected to enter button type as well.

File: apps/docs/src/app/core/component-docs/dialog/examples/auto-label/auto-label-dialog-example.component.html [hunk 8]
```diff
@@ -14,13 +14,15 @@ <h1 id="fd-dialog-header-10" fd-title fd-dialog-title>The History of Pineapple</
                 label="Interested"
                 [compact]="true"
                 (click)="dialog.close('Continue')"
+				ariaLabel="Interested emphasized"
```
Comment (kavya-b) [inline apps/docs/src/app/core/component-docs/dialog/examples/auto-label/auto-label-dialog-example.component.html:17]: please check formatting in this page

File: apps/docs/src/app/core/component-docs/dialog/examples/component-based/dialog-example.component.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (kavya-b) [reply apps/docs/src/app/core/component-docs/dialog/examples/component-based/dialog-example.component.ts:12]: i did not really understand why we are using `role="document"`. Which issue is this addressing?
From the [a11y ref link](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Document_Role#keyboard_interactions), it says `role="document"` should be used with `tabindex="0"` in order to read the static content. I don't see `tabindex` here, nor do I understand why we need `role="document"` in the first place: 
> Generally placed within an application role or other interactive widget role, the document role is used to indicate a section of a complex composite widget that an assistive technology user should read using its browse or virtual reading mode, if available.
Because ATs with reading mode default to that mode for all elements except for those with a widget or application role set, **document role is only useful for focusable elements within a widget or application that should be read as static rich text**.

File: apps/docs/src/app/core/component-docs/dialog/examples/dialog-state/dialog-state-example.component.html [hunk 9]
```diff
@@ -5,9 +5,7 @@ <h1 id='fd-dialog-header-7' fd-title>Dialog states</h1>
         </fd-dialog-header>
 
         <fd-dialog-body>
-			<div id='fd-dialog-body-7' role="document">
-				{{ dialog.data }}
-			</div>
+			<div id='fd-dialog-body-7' role="document">{{ dialog.data }}</div>
```
Comment (kavya-b) [inline apps/docs/src/app/core/component-docs/dialog/examples/dialog-state/dialog-state-example.component.html:8]:
```suggestion
			<div id="fd-dialog-body-7" role="document">{{ dialog.data }}</div>
```

File: libs/core/src/lib/dialog/base/dialog-config-base.class.ts [hunk 10]
```diff
@@ -29,6 +29,9 @@ export class DialogConfigBase<T> implements DynamicComponentConfig {
     /** Position of the dialog. */
     position?: DialogPosition;
 
+    /** Aria Model for the dialog component element */
```
Comment (kavya-b) [inline libs/core/src/lib/dialog/base/dialog-config-base.class.ts:32]:
```suggestion
    /** Aria Modal for the dialog component element */
```

File: libs/core/src/lib/dialog/base/dialog-content-base.class.ts [hunk 11]
```diff
@@ -9,6 +11,18 @@ export class DialogContentBase {
     /** Approve Button Label */
     approveButton?: string;
 
+    /** Aria Model for the dialog component element */
```
Comment (kavya-b) [inline libs/core/src/lib/dialog/base/dialog-content-base.class.ts:14]:
```suggestion
    /** Aria Modal for the dialog component element */
```

File: libs/core/src/lib/dialog/dialog-close-button/dialog-close-button.component.ts [hunk 12]
```diff
@@ -15,6 +15,7 @@ import { Component, Input } from '@angular/core';
         '[class.fd-button]': 'true',
         '[class.fd-button--compact]': '!mobile',
         '[class.fd-button--transparent]': 'true',
+        'title': 'close'
```
Comment (kavya-b) [inline libs/core/src/lib/dialog/dialog-close-button/dialog-close-button.component.ts:18]: what about i18n support for this static text?

File: libs/core/src/lib/dialog/dialog-default/dialog-default.component.html [hunk 13]
```diff
@@ -18,18 +19,22 @@ <h1 fd-title>{{ _defaultDialogContent?.title }}</h1>
...
-                [label]="_defaultDialogContent?.approveButton"
-                [compact]="!_defaultDialogConfiguration.mobile"
-                (click)="_approveButtonClicked()">
+            *ngIf="_defaultDialogContent?.approveButton"
+            fdType="emphasized"
+            fd-initial-focus
+            [label]="_defaultDialogContent?.approveButton"
+            [compact]="!_defaultDialogConfiguration.mobile"
+            (click)="_approveButtonClicked()"
+            [ariaLabel]="_defaultDialogContent?.approveButton + ' emphasized button'"
```
Comment (kavya-b) [inline libs/core/src/lib/dialog/dialog-default/dialog-default.component.html:28]: i18n support? it would be better if we could find a way not to hardcode this at all. `button` will get read by JAWS so it may not even be needed here.

File: libs/core/src/lib/dialog/dialog-default/dialog-default.component.html [hunk 14]
```diff
@@ -18,18 +19,22 @@ <h1 fd-title>{{ _defaultDialogContent?.title }}</h1>
...
         <fd-button-bar
-                *ngIf="_defaultDialogContent?.cancelButton"
-                [label]="_defaultDialogContent?.cancelButton"
-                [compact]="!_defaultDialogConfiguration.mobile"
-                (click)="_cancelButtonClicked()">
+            *ngIf="_defaultDialogContent?.cancelButton"
+            [label]="_defaultDialogContent?.cancelButton"
+            [compact]="!_defaultDialogConfiguration.mobile"
+            (click)="_cancelButtonClicked()"
+            [ariaLabel]="_defaultDialogContent?.cancelButton + ' transparent button'"
```
Comment (kavya-b) [inline libs/core/src/lib/dialog/dialog-default/dialog-default.component.html:36]: same comment as above

File: apps/docs/src/app/core/component-docs/dialog/dialog-docs.component.html [hunk 15]
```diff
@@ -230,20 +230,21 @@
 <ng-template let-dialog let-dialogConfig="dialogConfig" #playgroundDialog>
     <fd-dialog [dialogRef]="dialog" [dialogConfig]="dialogConfig">
         <fd-dialog-header>
-            <h1 fd-title>Playground Dialog</h1>
-            <button fd-dialog-close-button (click)="dialog.dismiss()"></button>
+            <h1 fd-title id="fd-dialog-header-11">Playground Dialog</h1>
+            <button fd-dialog-close-button (click)="dialog.dismiss()" i18n-title="@@dialogPlayGround.closebutton.title"></button>
```
Comment (kavya-b) [inline apps/docs/src/app/core/component-docs/dialog/dialog-docs.component.html:234]: for the example, i don't think we need to give i18n-title. But, we need to have `title` here instead. 

I think what we discussed was since you cannot add i18n marker to the `host` property of `fd-dialog-close-button`, we should instead add the `title` and `i18n-title` to wherever it is used in the dialog component. It should be in the library components where `fd-dialog-close-button` is used, not in the examples. Translation of examples is something we are not handling as of now so it is not needed here.

Also, you cannot have just `i18n-title` here and `title` in the library. We can't expect user to do this wherever they use `fd-dialog-close-button`, right? Therefore, both `title` and `i18n-title` should come from the library itself. 
Hope this is clear.

File: apps/docs/src/app/core/component-docs/dialog/examples/component-based/dialog-example.component.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (kavya-b) [reply apps/docs/src/app/core/component-docs/dialog/examples/component-based/dialog-example.component.ts:12]: you haven't addressed this. Please address or respond to all comments before asking for re-review.

File: apps/docs/src/app/core/component-docs/dialog/examples/dialog-backdrop-container/dialog-backdrop-container-example.component.html [hunk 16]
```diff
@@ -18,13 +18,11 @@
     <fd-dialog [dialogRef]="dialog" [dialogConfig]="dialogConfig">
         <fd-dialog-header>
             <h1 id="fd-dialog-header-2" fd-title>Backdrop Configuration</h1>
-            <button fd-dialog-close-button (click)="dialog.dismiss()"></button>
+            <button fd-dialog-close-button (click)="dialog.dismiss()" i18n-title="@@dialogPlayGround.closebutton.title"></button>
```
Comment (kavya-b) [inline apps/docs/src/app/core/component-docs/dialog/examples/dialog-backdrop-container/dialog-backdrop-container-example.component.html:21]: `i18n-title` not needed here as explained above. `title` is needed.

File: apps/docs/src/app/core/component-docs/dialog/examples/stacked-dialogs/first-dialog-example.component.ts [hunk 17]
```diff
@@ -7,7 +7,7 @@ import { SecondDialogExampleComponent } from './second-dialog-example.component'
         <fd-dialog>
             <fd-dialog-header>
                 <h1 id="fd-dialog-header-8" fd-title>First Dialog</h1>
-                <button fd-dialog-close-button (click)="dialogRef.dismiss('x')"></button>
+                <button fd-dialog-close-button (click)="dialogRef.dismiss('x')" i18n-title="@@dialogPlayGround.closebutton.title"></button>
```
Comment (kavya-b) [inline apps/docs/src/app/core/component-docs/dialog/examples/stacked-dialogs/first-dialog-example.component.ts:10]: `i18n-title` not needed here as explained above. `title` is needed.

File: apps/docs/src/app/core/component-docs/dialog/examples/stacked-dialogs/second-dialog-example.component.ts [hunk 18]
```diff
@@ -6,7 +6,7 @@ import { DialogRef } from '@fundamental-ngx/core/dialog';
         <fd-dialog>
             <fd-dialog-header>
                 <h1 id="fd-dialog-header-9" fd-title>Second Dialog</h1>
-                <button fd-dialog-close-button (click)="dialogRef.dismiss()"></button>
+                <button fd-dialog-close-button (click)="dialogRef.dismiss()" i18n-title="@@dialogPlayGround.closebutton.title"></button>
```
Comment (kavya-b) [inline apps/docs/src/app/core/component-docs/dialog/examples/stacked-dialogs/second-dialog-example.component.ts:9]: `i18n-title` not needed here as explained above. `title` is needed.

File: libs/core/src/lib/bar/button-bar/button-bar.component.ts [hunk 19]
```diff
@@ -24,7 +24,7 @@ import { ContentDensityService } from '@fundamental-ngx/core/utils';
               [fdType]="fdType"
               [label]="label"
               [attr.title]="ariaLabel"
-              [attr.aria-label]="ariaLabel"
+              [ariaLabel]="ariaLabel"
```
Comment (kavya-b) [inline libs/core/src/lib/bar/button-bar/button-bar.component.ts:27]: why was this changed? We are working on the native button so `[attr.aria-label]` should work fine. You can just pass `fdType` here itself, if it was specified.

File: libs/core/src/lib/button/base-button.ts [hunk 20]
```diff
@@ -61,6 +61,10 @@ export class BaseButton {
     @Input()
     fdMenu = false;
 
+    /** Sets aria-label attribute for button */
+    @Input()
+    ariaLabel: string;
```
Comment (kavya-b) [inline libs/core/src/lib/button/base-button.ts:66]: i don't think this is needed. See above comment.

File: libs/core/src/lib/button/button.component.ts [hunk 21]
```diff
@@ -37,7 +37,8 @@ import { applyCssClass } from '@fundamental-ngx/core/utils';
     changeDetection: ChangeDetectionStrategy.OnPush,
     host: {
         '[attr.type]': 'type',
-        '[attr.disabled]': '_disabled || null'
+        '[attr.disabled]': '_disabled || null',
+        '[attr.aria-label]': 'ariaLabel + " " + fdType'
```
Comment (kavya-b) [inline libs/core/src/lib/button/button.component.ts:41]: i don't think this is needed. See above comment on `button-bar`. Also, this does not check for undefined `fdType` cases.
image

File: libs/core/src/lib/dialog/dialog-default/dialog-default.component.html (hunk 14 above)
Comment (Lokanathan-k) [reply libs/core/src/lib/dialog/dialog-default/dialog-default.component.html]: fixed

File: libs/core/src/lib/dialog/dialog-default/dialog-default.component.html (hunk 13 above)
Comment (Lokanathan-k) [reply libs/core/src/lib/dialog/dialog-default/dialog-default.component.html]: fixed

File: libs/core/src/lib/dialog/dialog-close-button/dialog-close-button.component.ts (hunk 12 above)
Comment (Lokanathan-k) [reply libs/core/src/lib/dialog/dialog-close-button/dialog-close-button.component.ts:18]: modified

File: libs/core/src/lib/dialog/base/dialog-content-base.class.ts (hunk 11 above)
Comment (Lokanathan-k) [reply libs/core/src/lib/dialog/base/dialog-content-base.class.ts]: fixed

File: libs/core/src/lib/dialog/base/dialog-config-base.class.ts (hunk 10 above)
Comment (Lokanathan-k) [reply libs/core/src/lib/dialog/base/dialog-config-base.class.ts]: fixed

File: apps/docs/src/app/core/component-docs/dialog/examples/dialog-state/dialog-state-example.component.html (hunk 9 above)
Comment (Lokanathan-k) [reply apps/docs/src/app/core/component-docs/dialog/examples/dialog-state/dialog-state-example.component.html]: fixed

File: libs/core/src/lib/dialog/dialog-close-button/dialog-close-button.component.ts (hunk 12 above)
Comment (kavya-b) [reply libs/core/src/lib/dialog/dialog-close-button/dialog-close-button.component.ts:18]: as mentioned above, `title` should not be hardcoded here if you cannot also get `i18n-title` here.

**Review by kavya-b — COMMENTED** [review]
Please always mark what comments were addressed and if something is not, reply to it with your justification. It gets really difficult to track what comments were addressed and what were not and why.

adding ariba team as reviewer again for someone else to review this PR during my leave.

File: libs/core/src/lib/bar/button-bar/button-bar.component.ts (hunk 19 above)
Comment (Lokanathan-k) [reply libs/core/src/lib/bar/button-bar/button-bar.component.ts:27]: on introducing the change only in the button bar component will introduce a breaking change since it extends the baseButton class, in order to accomodate this breaking chagne this change was necessary

File: libs/core/src/lib/button/base-button.ts (hunk 20 above)
Comment (Lokanathan-k) [reply libs/core/src/lib/button/base-button.ts:66]: comments here https://github.com/SAP/fundamental-ngx/pull/6297#discussion_r690258903

File: libs/core/src/lib/button/button.component.ts (hunk 21 above)
Comment (Lokanathan-k) [reply libs/core/src/lib/button/button.component.ts:41]: fixed

File: apps/docs/src/app/core/component-docs/dialog/examples/stacked-dialogs/second-dialog-example.component.ts (hunk 18 above)
Comment (Lokanathan-k) [reply apps/docs/src/app/core/component-docs/dialog/examples/stacked-dialogs/second-dialog-example.component.ts]: fixed

File: apps/docs/src/app/core/component-docs/dialog/dialog-docs.component.html (hunk 15 above)
Comment (Lokanathan-k) [reply apps/docs/src/app/core/component-docs/dialog/dialog-docs.component.html]: fixed

File: apps/docs/src/app/core/component-docs/dialog/examples/dialog-backdrop-container/dialog-backdrop-container-example.component.html (hunk 16 above)
Comment (Lokanathan-k) [reply apps/docs/src/app/core/component-docs/dialog/examples/dialog-backdrop-container/dialog-backdrop-container-example.component.html]: fixed

File: apps/docs/src/app/core/component-docs/dialog/examples/stacked-dialogs/first-dialog-example.component.ts (hunk 17 above)
Comment (Lokanathan-k) [reply apps/docs/src/app/core/component-docs/dialog/examples/stacked-dialogs/first-dialog-example.component.ts]: fixed

File: apps/docs/src/app/core/component-docs/dialog/dialog-docs.component.html (hunk 15 above)
Comment (kavya-b) [reply apps/docs/src/app/core/component-docs/dialog/dialog-docs.component.html]: `i18n-title` is not needed for example files.

File: libs/core/src/lib/dialog/dialog-close-button/dialog-close-button.component.ts [hunk 22]
```diff
@@ -15,12 +15,16 @@ import { Component, Input } from '@angular/core';
...
 export class DialogCloseButtonComponent {
-
     /** Displays dialog close button in mobile mode */
     @Input()
     mobile = false;
+
+    /** add title dynamically to add a tooltip */
+    @Input()
+    title: string;
 }
```
Comment (Lokanathan-k) [inline libs/core/src/lib/dialog/dialog-close-button/dialog-close-button.component.ts:30]: introduced this as the title and aria lable was read twice

File: apps/docs/src/app/core/component-docs/dialog/examples/component-based/dialog-example.component.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (Lokanathan-k) [reply apps/docs/src/app/core/component-docs/dialog/examples/component-based/dialog-example.component.ts]: this cannot be removed, on checking with Accessability team they raised the issue to read the content as details, details in the sense the static content inside the dialog body.

File: libs/core/src/lib/button/base-button.ts [hunk 23]
```diff
@@ -77,9 +77,27 @@ export class BaseButton {
...
     /** @hidden */
     _disabled: boolean;
 
     /** @hidden */
     _ariaDisabled: boolean;
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        const ariaLabelChange = this.ariaLabel === undefined ? this.label : this.ariaLabel;
+        this._ariaLabel = ariaLabelChange + ' ' + this.fdType;
```
Comment (mikerodonnell89) [inline libs/core/src/lib/button/base-button.ts:101]: Appending the `this.fdType` works for English but will not translate to other languages.  We need to provide a way for the application developer to set the entire label and adequately communicate in the documentation that the responsibility to handle translations is on them because it is out of the hands of the library

**Comment by mikerodonnell89** [comment]
When I open a dialog, the buttons in the footer are focused and Chrome screenreader reads those, rather than the dialog title, which is what should be done according to issue 3 in #6100

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
see above comments

File: apps/docs/src/app/core/component-docs/dialog/examples/component-based/dialog-example.component.ts [hunk 24]
```diff
@@ -24,14 +24,16 @@ import { DialogRef } from '@fundamental-ngx/core/dialog';
                         label="Interesting"
                         fdType="emphasized"
                         [compact]="true"
-                        (click)="this.dialogRef.close('Continue')">
+                        (click)="this.dialogRef.close('Continue')"
```
Comment (Betrozov) [inline apps/docs/src/app/core/component-docs/dialog/examples/component-based/dialog-example.component.ts:27]: please remove `this`

File: apps/docs/src/app/core/component-docs/dialog/examples/component-based/dialog-example.component.ts [hunk 25]
```diff
@@ -24,14 +24,16 @@ import { DialogRef } from '@fundamental-ngx/core/dialog';
...
+                        ariaLabel="Interesting">
                 </fd-button-bar>
                 <fd-button-bar
                         label="Cancel"
                         fdInitialFocus
                         fdType="transparent"
                         [compact]="true"
-                        (click)="this.dialogRef.dismiss('Cancel')">
+                        (click)="this.dialogRef.dismiss('Cancel')"
```
Comment (Betrozov) [inline apps/docs/src/app/core/component-docs/dialog/examples/component-based/dialog-example.component.ts:35]: same

File: apps/docs/src/app/core/component-docs/dialog/examples/dialog-inner-popover/dialog-inner-popover.component.ts [hunk 26]
```diff
@@ -91,14 +91,16 @@ export class DialogInnerPopoverComponent {
                         label="Interesting"
                         fdType="emphasized"
                         [compact]="true"
-                        (click)="this.dialogRef.close('Interesting')">
+                        (click)="this.dialogRef.close('Interesting')"
```
Comment (Betrozov) [inline apps/docs/src/app/core/component-docs/dialog/examples/dialog-inner-popover/dialog-inner-popover.component.ts:94]: same

File: apps/docs/src/app/core/component-docs/dialog/examples/dialog-inner-popover/dialog-inner-popover.component.ts [hunk 27]
```diff
@@ -91,14 +91,16 @@ export class DialogInnerPopoverComponent {
...
+                        ariaLabel="Interesting">
                 </fd-button-bar>
                 <fd-button-bar
                         label="Cancel"
                         fdInitialFocus
                         fdType="transparent"
                         [compact]="true"
-                        (click)="this.dialogRef.dismiss('Cancel')">
+                        (click)="this.dialogRef.dismiss('Cancel')"
```
Comment (Betrozov) [inline apps/docs/src/app/core/component-docs/dialog/examples/dialog-inner-popover/dialog-inner-popover.component.ts:102]: same

**Review by Betrozov — CHANGES_REQUESTED** [review]

**Comment by Lokanathan-k** [comment]
> When I open a dialog, the buttons in the footer are focused and Chrome screenreader reads those, rather than the dialog title, which is what should be done according to issues 3 and 5 in #6100

> Dialog title is read in Jaws and A11y supported is offered only to Jaws
Screenshot 2021-08-19 at 9 45 04 AM

File: apps/docs/src/app/core/component-docs/dialog/dialog-docs.component.html [hunk 28]
```diff
@@ -230,20 +230,21 @@
 <ng-template let-dialog let-dialogConfig="dialogConfig" #playgroundDialog>
     <fd-dialog [dialogRef]="dialog" [dialogConfig]="dialogConfig">
         <fd-dialog-header>
-            <h1 fd-title>Playground Dialog</h1>
-            <button fd-dialog-close-button (click)="dialog.dismiss()"></button>
+            <h1 fd-title id="fd-dialog-header-11">Playground Dialog</h1>
+            <button fd-dialog-close-button (click)="dialog.dismiss()" title="close" i18n-title="@@dialogPlayGround.closebutton.title"></button>
```
Comment (Betrozov) [inline apps/docs/src/app/core/component-docs/dialog/dialog-docs.component.html:234]: you do not need i18n in docs app

File: apps/docs/src/app/core/component-docs/dialog/examples/dialog-backdrop-container/dialog-backdrop-container-example.component.html [hunk 29]
```diff
@@ -18,13 +18,11 @@
     <fd-dialog [dialogRef]="dialog" [dialogConfig]="dialogConfig">
         <fd-dialog-header>
             <h1 id="fd-dialog-header-2" fd-title>Backdrop Configuration</h1>
-            <button fd-dialog-close-button (click)="dialog.dismiss()"></button>
+            <button fd-dialog-close-button (click)="dialog.dismiss()" title="close" i18n-title="@@dialogPlayGround.closebutton.title"></button>
```
Comment (Betrozov) [inline apps/docs/src/app/core/component-docs/dialog/examples/dialog-backdrop-container/dialog-backdrop-container-example.component.html:21]: same

File: apps/docs/src/app/core/component-docs/dialog/examples/stacked-dialogs/first-dialog-example.component.ts [hunk 30]
```diff
@@ -7,11 +7,11 @@ import { SecondDialogExampleComponent } from './second-dialog-example.component'
         <fd-dialog>
             <fd-dialog-header>
                 <h1 id="fd-dialog-header-8" fd-title>First Dialog</h1>
-                <button fd-dialog-close-button (click)="dialogRef.dismiss('x')"></button>
+                <button fd-dialog-close-button (click)="dialogRef.dismiss('x')" title="close" i18n-title="@@dialogPlayGround.closebutton.title"></button>
```
Comment (Betrozov) [inline apps/docs/src/app/core/component-docs/dialog/examples/stacked-dialogs/first-dialog-example.component.ts:10]: same

File: apps/docs/src/app/core/component-docs/dialog/examples/stacked-dialogs/second-dialog-example.component.ts [hunk 31]
```diff
@@ -6,11 +6,11 @@ import { DialogRef } from '@fundamental-ngx/core/dialog';
         <fd-dialog>
             <fd-dialog-header>
                 <h1 id="fd-dialog-header-9" fd-title>Second Dialog</h1>
-                <button fd-dialog-close-button (click)="dialogRef.dismiss()"></button>
+                <button fd-dialog-close-button (click)="dialogRef.dismiss()" title="close" i18n-title="@@dialogPlayGround.closebutton.title"></button>
```
Comment (Betrozov) [inline apps/docs/src/app/core/component-docs/dialog/examples/stacked-dialogs/second-dialog-example.component.ts:9]: same

File: libs/core/src/lib/button/base-button.ts [hunk 32]
```diff
@@ -1,4 +1,4 @@
-import { Directive, Input } from '@angular/core';
+import { AfterViewInit, Directive, Inject, Input, LOCALE_ID } from '@angular/core';
```
Comment (Betrozov) [inline libs/core/src/lib/button/base-button.ts]: if you not use LOCALE_ID than please remove it

File: libs/core/src/lib/button/base-button.ts [hunk 33]
```diff
@@ -77,6 +77,9 @@ export class BaseButton {
 
     set ariaDisabled(value: boolean) { this._ariaDisabled = coerceBooleanProperty(value); }
 
+    /** @hidden */
+    private _ariaLabel: string;
```
Comment (Betrozov) [inline libs/core/src/lib/button/base-button.ts:81]: same for it

File: libs/core/src/lib/button/base-button.ts [hunk 34]
```diff
@@ -1,4 +1,4 @@
-import { Directive, Input } from '@angular/core';
+import { AfterViewInit, Directive, Inject, Input } from '@angular/core';
```
Comment (Betrozov) [inline libs/core/src/lib/button/base-button.ts:1]: if you do not use AfterViewInit and Inject then please remove it

**Review by Betrozov — CHANGES_REQUESTED** [review]

**Review by mikerodonnell89 — APPROVED** [review]

**Review by mikerodonnell89 — COMMENTED** [review]
> > Dialog title is read in Jaws and A11y supported is offered only to Jaws

Are you saying a11y is only supported through jaws? A windows-only screenreader? That doesn't sound right to me

**Comment by Lokanathan-k** [comment]
> > > Dialog title is read in Jaws and A11y supported is offered only to Jaws
> 
> Are you saying a11y is only supported through jaws? A windows-only screenreader? That doesn't sound right to me

> yes, thats the limitation of jaws, when ever there there is a focus on the elements inside the dialog reads the title twice. this is already informed to @manjunathanagaraj  about this. and decided we support only thru Jaws.

File: apps/docs/src/app/core/component-docs/dialog/examples/auto-label/auto-label-dialog-example.component.html (hunk 6 above)
Comment (droshev) [reply apps/docs/src/app/core/component-docs/dialog/examples/auto-label/auto-label-dialog-example.component.html]: I am not sure how this satisfy any accessibility requirement to say the type of the button(e.g. transparent, ghost, default). @Lokanathan-k can you ask which ACC requirement is covering this?

Comment (droshev) [reply apps/docs/src/app/core/component-docs/dialog/examples/auto-label/auto-label-dialog-example.component.html]: > yes, it has been raised as issues by the accessibility team. the changes has been introduced post verifying with the team.

I am not sure how this satisfy any accessibility requirement to say the type of the button(e.g. transparent, ghost, default). @Lokanathan-k can you ask which ACC requirement is covering this?

**Comment by droshev** [comment]
> when ever there there is a focus on the elements inside the dialog

@Lokanathan-k is there a document or other documentation explaining this? I don't think we should limit our support to JAWS only. Maybe I didn't understand that correctly?

File: apps/docs/src/app/core/component-docs/dialog/examples/auto-label/auto-label-dialog-example.component.html (hunk 6 above)
Comment (Lokanathan-k) [reply apps/docs/src/app/core/component-docs/dialog/examples/auto-label/auto-label-dialog-example.component.html]: @droshev the requirement from ACC team is to diffrentate the normal button and the emphasied button. and ACC team requires the fdtype="emphasised" to be read, and the chagne now has been introduced to handle it from arial label.

**Comment by puru-hk** [comment]
Looks good

**Review by puru-hk — APPROVED** [review]

**Review by platon-rov — APPROVED** [review]
